iOS/RCTOrientation/Orientation.m

```diff
@@ -20,6 +21,32 @@ + (UIInterfaceOrientationMask)getOrientation {
   return _orientation;
 }
 
+- (void)updateInterfaceOrientation: (UIInterfaceOrientation)orientation
+               withOrientationMask: (UIInterfaceOrientationMask) orientationMask
+                           resolve: (RCTPromiseResolveBlock) resolve
+                            reject: (RCTPromiseRejectBlock)reject
+{
+  [[NSOperationQueue mainQueue] addOperationWithBlock:^ {
+    if (@available(iOS 16.0, *)) {
+      NSArray<UIScene *> *connectedScenes = [[[UIApplication sharedApplication] connectedScenes] allObjects];
+      if ([connectedScenes count] > 0) {
+        UIWindowScene *windowScene = (UIWindowScene *) connectedScenes[0];
+        [windowScene requestGeometryUpdateWithPreferences: [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:orientationMask]
```

Comment:
TODO: I think we need to call `setNeedsUpdateOfSupportedInterfaceOrientations` before calling `requestGeometryUpdateWithPreferences`. I think Neal tried this out locally already though, so there may still be issues that won't get fixed with only that change

Reply:
So after digging into this a bit more, I understand the usage of `setNeedsUpdateOfSupportedInterfaceOrientations`. 

`setNeedsUpdateOfSupportedInterfaceOrientations` trigger a rotation change if the supported interface orientations change (which they do when we do `[Orientation setOrientation]`)

So generally it's like a 2 step process:
1) Try to request a geometry update which will rotate the screen. This will fail if the supported interfaces do not support the requested orientation mask. (i.e if I am locked to landscape right, there will be an error coming from the callback indicating that the orientation change request failed)
2) Then we do a `setNeedsUpdateOfSupportedInterfaceOrientations`, in the case where step 1 fails, this will re-orient the screen based on the supported interfaces

Reply:
Thanks for the explanation! A note for other reviewers: this PR is now always calling `setNeedsUpdateOfSupportedInterfaceOrientations` after we call `requestGeometryUpdateWithPreferences`.